### PR TITLE
prov/efa: Add tracing point for unexpected recv entering the queue

### PR DESCRIPTION
**Pull request by rauteric**
The current code includes a trace when an unexpected message is processed, but not when it first enters the queue. This patch adds the extra trace.

Signed-off-by: Eric Raut <eraut@amazon.com>

### CODE REVIEW

**Comment by wzamazon** [comment]
bot:aws:retest

File: prov/efa/src/rxr/rxr_pkt_type_req.c [hunk 1]
```diff
@@ -1204,6 +1204,9 @@ struct rxr_op_entry *rxr_pkt_get_tagrtm_rx_entry(struct rxr_ep *ep,
 			efa_eq_write_error(&ep->util_ep, FI_ENOBUFS, FI_EFA_ERR_RX_ENTRIES_EXHAUSTED);
 			return NULL;
 		}
+		rxr_tracing(msg_recv_unexpected, rx_entry->msg_id,
```
Comment (wzamazon) [inline prov/efa/src/rxr/rxr_pkt_type_req.c:1207]: The event name is a little ambiguous. It should have `tagged` in it to indiciate the message is ad tagged message. Probably something like `uexpected_tagged_message_received`

Comment (wzamazon) [reply prov/efa/src/rxr/rxr_pkt_type_req.c:1207]: Can you also add an trace in `rxr_pkt_get_msgrtm_rx_entry` for non-tagged version of this event.

**Review by shefty — CHANGES_REQUESTED** [review]
Can you please update the commit subject to indicate 'prov/efa'?  Thanks

File: prov/efa/src/rxr/rxr_pkt_type_req.c (hunk 1 above)
Comment (rauteric) [reply prov/efa/src/rxr/rxr_pkt_type_req.c]: I named the events msg_recv_unexpected_{tagged/untagged} to match the style of the other events.

**Comment by rauteric** [comment]
The build is currently failing for me with LTTNG enabled, but I don't think it's related to my change:
```
prov/efa/src/rxr/rxr_pkt_entry.c: In function 'rxr_pkt_entry_recv':
prov/efa/src/rxr/rxr_pkt_entry.c:484:3: warning: implicit declaration of function 'efa_tracing' [-Wimplicit-function-declaration]
   efa_tracing(post_recv, (void *) head->wr_id);
   ^~~~~~~~~~~
prov/efa/src/rxr/rxr_pkt_entry.c:484:15: error: 'post_recv' undeclared (first use in this function); did you mean 'ibv_post_recv'?
   efa_tracing(post_recv, (void *) head->wr_id);
               ^~~~~~~~~
               ibv_post_recv
prov/efa/src/rxr/rxr_pkt_entry.c:484:15: note: each undeclared identifier is reported only once for each function it appears in
```

**Comment by wzamazon** [comment]
@rauteric 

Yes. The building failure is not due to your change.

**Comment by wzamazon** [comment]
@rauteric 

https://github.com/ofiwg/libfabric/pull/8398 should fix the compilation issue

**Review by wzamazon — APPROVED** [review]

**Comment by wzamazon** [comment]
@shefty 

I can see @rauteric made the change you requested. So I will merge

**Pull request merged**